spidermon/contrib/validation/utils.py

```diff
@@ -28,3 +36,43 @@ def is_valid_url(url):
 
 def is_valid_email(email):
     return not EMAIL_REGEX.match(email) is None
+
+
+def get_schema_from(source):
+    if is_schema_url(source):
+        schema = get_contents(source)
+        try:
+            return json.loads(schema)
+        except Exception as e:
+            logger.exception(
+                str(e) + "\nCould not parse schema from '{}'".format(source)
+            )
+    elif source.endswith(".json"):
```

Comment:
This requirement, the file name having to have a `json` extension when it is a local file, should be clear from the documentation.

Reply:
Agreed, will add this bit.